src/scheduler.rs

```diff
@@ -1,33 +1,71 @@
+use chrono::{DurationRound, TimeDelta, TimeZone, Utc};
 use std::sync::Arc;
 
-use chrono::Utc;
 use tokio::sync::mpsc;
+
+use std::time::Duration;
 use tokio::task::JoinHandle;
-use tokio::time::{self, Instant};
+use tokio::time::{interval, Instant};
+
 use tokio_util::sync::CancellationToken;
 
 use crate::check_executor::{queue_check, CheckSender};
 use crate::config_store::{RwConfigStore, Tick};
+use redis::{AsyncCommands, Client};
 
 pub fn run_scheduler(
     partition: u16,
     config_store: Arc<RwConfigStore>,
     executor_sender: CheckSender,
     shutdown: CancellationToken,
+    progress_key: String,
+    redis_host: String,
 ) -> JoinHandle<()> {
     tracing::info!(partition, "scheduler.starting");
-    tokio::spawn(async move { scheduler_loop(config_store, executor_sender, shutdown).await })
+    tokio::spawn(async move {
+        scheduler_loop(
+            config_store,
+            executor_sender,
+            shutdown,
+            progress_key,
+            redis_host,
+        )
+        .await
+    })
 }
 
 async fn scheduler_loop(
     config_store: Arc<RwConfigStore>,
     executor_sender: CheckSender,
     shutdown: CancellationToken,
+    progress_key: String,
+    redis_host: String,
 ) {
-    let mut interval = time::interval(time::Duration::from_secs(1));
+    let client = Client::open(redis_host.clone()).unwrap();
+    let mut connection = client
+        .get_multiplexed_tokio_connection()
+        .await
+        .expect("Unable to connect to Redis");
+    let result: Option<String> = connection
+        .get(&progress_key)
+        .await
+        .expect("Unable to get last tick key");
+    let tick_frequency = Duration::from_secs(1);
+    tracing::debug!(progress_key, result, "scheduler.redis_stored_tick_value");
+    let start = match result {
+        Some(result) => {
+            let nanos: i64 = result.parse().unwrap_or(Utc::now().timestamp());
+            Utc.timestamp_nanos(nanos) + tick_frequency
```

Comment:
We add one since this was the most recent completed check right. maybe a better name would be

```suggestion
            let last_completed_check_nanos: i64 = result.parse().unwrap_or(Utc::now().timestamp());
            Utc.timestamp_nanos(last_completed_check_nanos) + tick_frequency
```